Lib/test/test_json/test_speedups.py

```diff
@@ -36,6 +37,26 @@ def test_make_encoder(self):
             b"\xCD\x7D\x3D\x4E\x12\x4C\xF9\x79\xD7\x52\xBA\x82\xF2\x27\x4A\x7D\xA0\xCA\x75",
```

Comment:
This isn't related to this issue, but it seems to me that TypeError is raised for different cause than was originally in this test. This test doesn't work as intended.

Reply:
IIUC, Victor wrote this test to verify that the interpreter doesn't crash (as part of https://bugs.python.org/issue6986, as originally `PyArg_ParseTupleAndKeywords()` was given actual fields of the new `PyEncoderObject`, instead of temp vars), so raising a TypeError is the wanted outcome here.
maybe we can just change the name of the test to 'test_issue6986' (as it tests only this specific issue)?
and maybe do the same for `test_make_scanner()`?
anyway, I guess any such change should be in a PR referring to bpo-6986..

Reply:
Seems your are right. Thank you for a reference.